src/Site/Key.php

```diff
@@ -74,5 +74,27 @@ public static function generate(int $length = 32): string
     public static function generateEncode(int $length = 32): string
     {
         return static::encode(self::generate($length));
-    }
+	}
+
+	/**
+	 * Generate RSA public and private keypair.
+	 *
+     * @since 3.0.0
+     *
+	 * @return array
+	 */
+	public static function generateRSAKey(): array
+	{
+		$rsa = openssl_pkey_new(array(
+			"private_key_bits" => 2048,
+			"private_key_type" => \OPENSSL_KEYTYPE_RSA,
+		));
+		openssl_pkey_export($rsa, $rsaPrivateKey);
+		$details = openssl_pkey_get_details($rsa);
```

Comment:
Please check this [post](https://paragonie.com/blog/2018/04/protecting-rsa-based-protocols-against-adaptive-chosen-ciphertext-attacks#rsa-doing-it-right) to concern how to generate RSA key correctly.